app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsPagingSource.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>.
+ */
+
+package com.keylesspalace.tusky.components.notifications
+
+import android.util.Log
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.keylesspalace.tusky.entity.Notification
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.HttpHeaderLink
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import okhttp3.Headers
+import retrofit2.Response
+import javax.inject.Inject
+
+/** Models next/prev links from the "Links" header in an API response */
+data class Links(val next: String?, val prev: String?)
+
+/** [PagingSource] for Mastodon Notifications, identified by the Notification ID */
+class NotificationsPagingSource @Inject constructor(
+    private val mastodonApi: MastodonApi,
+    private val notificationFilter: Set<Notification.Type>
+) : PagingSource<String, Notification>() {
+    override suspend fun load(params: LoadParams<String>): LoadResult<String, Notification> {
+        Log.d(TAG, "load() with ${params.javaClass.simpleName} for key: ${params.key}")
```

Comment:
Do we need/want these debug statements (also elsewhere)?

Reply:
I prefer to keep them in. They're completely stripped in release builds, and are helpful for debugging, especially in asynchronous code.

Sometimes setting a breakpoint and inspecting values at a particular point in time is what you want, and other times you want to see what code gets hit and in what order.